### DISCUSSION TITLE
Multi bucket with multi filer

### DISCUSSION THREAD

**Discussion by SmsS4** [seaweedfs/seaweedfs#4463] [Q&A] [answered]
Hello
I have multiple filer with different filer store.  
So I can't have two bucket with same name in two filer because they use same collection name.  
What is best approach to have multiple bucket with same name on different filers?  
The only solution I found is to use a random number (or maybe filerGroup) as prefix for collection name (for example `filerGroup-bucketName` for collection name).  
Is there a better way to achieve this? and if I implement this feature does is get merge?

**Comment by chrislusf** [comment, marked as answer]
filerGroup is a good choice.

**Reply by SmsS4** [reply]
Can you give me a hint for implementing this feature?
I looked at the code but didn't understand when and where Seaweed creates collection
